src/main/java/com/nvidia/spark/rapids/jni/BloomFilter.java

```diff
@@ -0,0 +1,164 @@
+    }
+    if(buffer.getLength() != bloomFilterByteSize(bloomFilterBits)){
+      throw new IllegalArgumentException("Invalid pre-existing buffer passed. Size mismatch");
+    }
+
+    this.numHashes = numHashes;
+    this.bloomFilterBits = bloomFilterBits;
+    bloomFilter = buffer;
+  }
+
+  @Override
+  public void close() {
+    bloomFilter.close();
+  }
+
+  /**
+   * Insert a column of longs into the bloom filter.
+   * @param cv The column containing the values to add.
+   */
+  public void put(ColumnView cv){
+    put(bloomFilter.getAddress(), bloomFilter.getLength(), bloomFilterBits, cv.getNativeView(), numHashes);
+  }
+
+  /**
+   * Probe the bloom filter with a column of longs. Returns a column of booleans. For 
+   * each row in the output; a value of true indicates that the corresponding input value
+   * -may- be in the set of values used to build the bloom filter; a value of false indicates
+   * that the corresponding input value is conclusively not in the set of values used to build
+   * the bloom filter.
+   * @param cv The column containing the values to check.
+   * @return A boolean column indicating the results of the probe.
+   */
+  public ColumnVector probe(ColumnView cv){
+     return new ColumnVector(probe(cv.getNativeView(), bloomFilter.getAddress(), bloomFilter.getLength(), bloomFilterBits, numHashes));
+  }
+
+  /**
+   * Retrieve the underlying device memory buffer
+   * @return The buffer containing the bloom filter.
+   */
+  public DeviceMemoryBuffer getBuffer(){
+    return bloomFilter;
+  }
+
+  /**
+   * Merge one or more bloom filters into a new bloom filter.
+   * @param bloomFilters The bloom filters to be merged
+   * @return A new bloom filter containing the merged inputs.
+   */
+  public static BloomFilter merge(BloomFilter bloomFilters[]){
+    if(bloomFilters.length < 1){
+      throw new IllegalArgumentException("Must pass at least 1 bloom filters to merge");
+    }    
+    // verify all the bloom filters match in size
+    for (int i = 1; i < bloomFilters.length; i++) {
+      if(bloomFilters[i].numHashes != bloomFilters[0].numHashes){
+        throw new IllegalArgumentException("All bloom filters must use the same number of hashes");
+      }
+      if(bloomFilters[i].bloomFilterBits != bloomFilters[0].bloomFilterBits){
+        throw new IllegalArgumentException("All bloom filters must have the same number of bits");
+      }
+    }
+
+    long buffers[] = new long[bloomFilters.length];
+    for (int i = 0; i < bloomFilters.length; i++) {
+      buffers[i] = bloomFilters[i].getBuffer().getAddress();
+    }
+
+    int numHashes = bloomFilters[0].numHashes;
+    long bloomFilterBits = bloomFilters[0].bloomFilterBits;
+    long bloomFilterBytes = bloomFilterByteSize(bloomFilterBits);
+
+    long[] merged = merge(buffers, bloomFilterBytes);
+    return new BloomFilter(numHashes, bloomFilterBits, DeviceMemoryBuffer.fromRmm(merged[0], bloomFilterBytes, merged[1]));
```

Comment:
nit: Technically if we made a mistake and allocated the merged bit/bytes wrong, the resulting bloom filter would be leaked because an exception would be thrown inside the constructor and nothing would be closed/cleaned up.

This is very minor.

Reply:
I think what you want to do is the following.

```
long merged = merge(buffers, bloomFilterBytes);
DeviceMemoryBuffer buffer = fromRMM(merged[0], bloomFilterBytes, merged[1]);
try {
  return new BloomFilter(numHashes, bloomFilterBits, buffer);
} catch (Throwable t) {
  buffer.close();
  buffer = null;
  throw t;
}
```